Core/include/Acts/AmbiguityResolution/AthenaAmbiguityResolution.hpp

```diff
@@ -0,0 +1,223 @@
+          template <typename> class holder_t>
+using OptinalScoreModifier = std::function<void(
+    const Acts::TrackProxy<track_container_t, trajectory_t, holder_t, true>&,
+    double&)>;
+
+}  // namespace AthenaAmbiguitySolver
+
+namespace Acts {
+
+/// Generic implementation of the athena ambiguity resolution
+/// The alhorithm is based on the following steps:
+/// 1) Compute the initial state of the tracks
+/// 2) Compute the score of each track
+/// 3) Removes hits that are not good enough for each track
+/// 4) Remove tracks that have a score below a certain threshold or not have
+/// enough hits 5) Remove tracks that are not good enough based on cuts Contains
+/// method for data preparations
+class AthenaAmbiguityResolution {
+ public:
+  /// @brief Detector configuration struct : contains the configuration for each detector
+  ///
+  /// The configuration can be saved in a json file and loaded from there.
+  ///
+  struct DetectorConfig {
+    int hitsScoreWeight;
+    int holesScoreWeight;
+    int outliersScoreWeight;
+    int otherScoreWeight;
+
+    std::size_t minHits;
+    std::size_t maxHits;
+    std::size_t maxHoles;
+    std::size_t maxOutliers;
+    std::size_t maxSharedHits;
+
+    bool sharedHitsFlag;  // if true, the shared hits are considered as bad hits
+                          // for this detector
+
+    std::size_t detectorId;
+
+    std::vector<double>
+        factorHits;  // a list of values from  0 to 1, the higher number of
+                     // hits, higher value in the list is multiplied to
+                     // ambuiguity score
+                     // applied only if useAmbiguityFunction is true
+    std::vector<double>
+        factorHoles;  // a list of values from  0 to 1, the higher number of
+                      // holes, lower value in the list is multiplied to
+                      // ambuiguity score
+                      // applied only if useAmbiguityFunction is true
+  };
+
+  /// @brief  TrackFeatures struct : contains the features that are counted for each track.
+  ///
+  /// The trackFeatures is used to compute the score of each track
+  struct TrackFeatures {
+    std::size_t nHits;
+    std::size_t nHoles;
+    std::size_t nOutliers;
+    std::size_t nSharedHits;
+  };
+
+  /// @brief Configuration struct : contains the configuration for the ambiguity resolution.
+  struct Config {
+    std::map<std::size_t, std::size_t> volumeMap = {{0, 0}};
+    std::map<std::size_t, DetectorConfig> detectorMap;
+    // minimum score for any track
+    double minScore = 0;
+    // minimum score for shared tracks
+    double minScoreSharedTracks = 0;
+    // maximum number of shared tracks per measurement
+    std::size_t maxSharedTracksPerMeasurement = 10;
+    // maximum number of shared hit per track
+    std::size_t maxShared = 5;
+
+    double pTMin = 0;
+    double pTMax = 1e9;
+
+    double phiMin = -M_PI;
+    double phiMax = M_PI;
+
+    double etaMin = -5;
+    double etaMax = 5;
```

Comment:
This seems to be duplicated with `TrackSelector`. Consider reusing the track selector or expecting selected tracks

Reply:
After discussing with @CarloVarni , Alex and @Corentin-Allaire. we decided the entire scoring could be changed to be less Athena like and to make it more open-ended.  So that probably for  PR.